plugins/cros-ec/fu-cros-ec-usb-device.c

```diff
@@ -578,11 +815,54 @@ fu_cros_ec_usb_device_prepare_firmware (FuDevice *device,
 	return g_steal_pointer (&firmware);
 }
 
+static gboolean
+fu_cros_ec_usb_device_attach (FuDevice *device, GError **error)
+{
+	FuCrosEcUsbDevice *self = FU_CROS_EC_USB_DEVICE (device);
+
+	if (!self->in_bootloader) {
+		/* already in rw, so skip jump to rw */
+		return TRUE;
+	}
+
+	fu_device_set_remove_delay (device, FU_DEVICE_REMOVE_DELAY_RE_ENUMERATE);
+	if (!fu_cros_ec_usb_device_jump_to_rw (device, error)) {
+		return FALSE;
+	}
+
+	fu_device_set_status (device, FWUPD_STATUS_DEVICE_RESTART);
+	fu_device_add_flag (device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_cros_ec_usb_device_detach (FuDevice *self, GError **error)
+{
+	if (fu_device_has_custom_flag (self, "rw-written") &&
+	    !fu_device_has_custom_flag (self, "ro-written"))
+		return TRUE;
+
+	fu_device_set_remove_delay (self, FU_DEVICE_REMOVE_DELAY_RE_ENUMERATE);
+	if (!fu_cros_ec_usb_device_reset_to_ro (self, error)) {
+
+		return FALSE;
+	}
```

Comment:
style nit; can drop braces and whitespace here

Reply:
Done. Thanks.